src/core/internal/backtrace/libunwind.d

```diff
@@ -0,0 +1,153 @@
+/**
+ * Basic D language bindings for LLVM libunwind
+ *
+ * There are two available libunwind: The "upstream" one, inherited
+ * from HP, which is maintained as a GNU project,
+ * and the LLVM one, part of llvm-project, and the default on Mac OSX.
+ *
+ * They are both essential part of other languages ABI, and are available
+ * in both GCC and LLVM. However, in GCC, only the higher-level functions
+ * are exposed (e.g. `_Unwind_*`) while LLVM expose the higher-level
+ * and lower-level (`unw_*`) functions.
+ * Many distributions have a `libunwind` package as well, that provides
+ * the `unw_*` functions, but since it also supports remote unwinding,
+ * the function names are actually platform dependent and binding them
+ * is a pain as many things rely on `#define`.
+ *
+ * In the future, we would like to implement backtrace using only the
+ * higher-level functions (`_Unwind_*`), which will allow us to not
+ * use `backtrace` and friends directly, and only retrieve the functions
+ * names when needed (currently we need to eagerly get the functions names).
```

Comment:
Why aren't you using `_Unwind_Backtrace` from the start?

Reply:
Because I learned on the job, mostly. The original implementation aimed to be compatible with the old API (HP libunwind), which doesn't have those functions IIRC. Later I realized that just going with the GCC / LLVM libunwind was much better and ditched compatibility with HP libunwind. Note that this is only (or should only) be used on Alpine Linux at the moment.

Reply:
Original meaning 40+ year old API?  If I'm not mistaken the backtrace function has been around since 2000s.

Reply:
Something like this should suffice (though I didn't write it, and not sure I've ever tested it either given that libbacktrace is always present in gcc's tree).
https://github.com/gcc-mirror/gcc/blob/b2cd32b743ba440e75505ce30c6b5c592ed144ea/libphobos/libdruntime/gcc/backtrace.d#L428-L444